x-pack/auditbeat/module/system/socket/state.go

```diff
@@ -563,6 +563,38 @@ func (s *state) CreateProcess(p *process) error {
 	return nil
 }
 
+func (s *state) ForkProcess(parentPID, childPID uint32, ts kernelTime) error {
+	if parentPID == childPID {
+		return nil
+	}
+	s.Lock()
+	defer s.Unlock()
+	if _, found := s.processes[childPID]; found {
+		return errors.New("fork: child pid already registered to another process")
+	}
+	if parent, found := s.processes[parentPID]; found {
+		child := &process{
+			pid:         childPID,
+			name:        parent.name,
+			path:        parent.path,
+			args:        parent.args, // TODO: shallow copy
```

Comment:
Would this copy be useful? We don't mutate `process.args` element-wise after creation, so this should be fine the way you have it.

Reply:
Done